subsys/canbus/isotp_fast/isotp_fast_internal.h

```diff
@@ -86,3 +86,15 @@ struct isotp_fast_recv_await_ctx
     struct isotp_fast_recv_ctx *rctx;
 };
 #endif
+
+#ifdef CONFIG_ISOTP_FAST_FIXED_ADDRESSING
+
+/**
+ * Create the counterpart address from a given fixed address by swapping the target
+ * address and the source address.
+ *
+ * @returns resulting address
+ */
+struct isotp_fast_addr isotp_fast_get_tx_addr_fixed(const struct isotp_fast_addr *addr);
+
+#endif
```

Comment:
I guess this will eventually have to move into `isotp_fast.h` if it is to be merged into Zephyr (or we duplicate it in can.c). (It's a fairly useful utility function that other users of the ISO-TP stack will probably want to use if they are doing anything custom derived from ISO-TP fixed addressing.)

Reply:
My thoughts were that we allow to define a custom callback for custom addressing, but we use this one if fixed addressing is selected (which is defined by the standard and will not change). That's why I thought this should be put into the private header. We can revisit that in the future, though.